runtime/plugin/editorconfig.lua

```diff
@@ -0,0 +1,12 @@
+if vim.g.loaded_editorconfig then
+  return
+end
+vim.g.loaded_editorconfig = 1
```

Comment:
I know this is the longstanding convention for opt-ing out of builtin plugins, but is this really something we want to continue to perpetuate? 

Users doing `vim.g.loaded_editorconfig = 1` feels really hacky. It's like they're lying to nvim. Imagine if you're a new user and you see this in someone else's config? One could argue it looks like it's enabling something.

How about something like:

```lua
if vim.g.loaded_editorconfig or vim.g.disable_editorconfig then
```

It's maybe a lil redundant to have 2 variables, but I think it pays back in the UX.

Reply:
For non-vimscript plugins I think we should find a new convention. `loaded_foo` was a hack to prevent re-loading of `plugin/foo.vim` scripts, that gained this extra purpose.

- Perhaps vim needs a `:packnoload` command that disables `start/` plugins   (basically the inverse of `opt/`)? 
    -  No, that's too special-case: it would only be useful for builtin plugins, and currently there are zero `start/` plugins builtin.

Meanwhile, plugin settings are conventionally prefixed with the plugin name, right? `loaded_foo` was a special case. So if we use a global var, should probably be named `vim.g.editorconfig_enable`.

Reply:
Changed to `editorconfig_enable`.